driver-core/src/test/unit/com/mongodb/CustomMatchers.groovy

```diff
@@ -55,25 +58,32 @@ class CustomMatchers {
         if (actual.class.name != expected.class.name) {
             return false
         }
-        getFieldNames(actual.class).findAll { !ignoreNames.contains(it) } .collect {
-            if (nominallyTheSame(it)) {
-                return actual."$it".class == expected."$it".class
-            } else if (actual."$it" != expected."$it") {
-                def (a1, e1) = [actual."$it", expected."$it"]
-                if ([a1, e1].contains(null) && [a1, e1] != nullList) {
+        getFields(actual.class).findAll { !ignoreNames.contains(it.name) } .collect {
+            it.setAccessible(true)
+            def actualPropertyValue = it.get(actual)
```

Comment:
Access value by field rather than property.  This allows the values to be accessed directly rather than by property (essentially a public accessor method).  I made this change because there were a bunch of fields that did not have public accessor methods, which made them not accessible and thus this threw a MissingPropertyException without this change.